File: 2465-number-of-distinct-averages/2465-number-of-distinct-averages.java
class Solution {
    public int distinctAverages(int[] nums) {
        HashSet<Double> set=new HashSet<>();
        Arrays.sort(nums);
        int min=0;
        int max=nums.length-1;
        while(min<max){
            double avg=(nums[min]+nums[max])/2.0;
            set.add(avg);
            min++;
            max--;
        }
        return set.size();
    }
}

File: 2465-number-of-distinct-averages/2465-number-of-distinct-averages.cpp
class Solution {
public:
    int distinctAverages(vector<int>& nums) {
         sort(nums.begin(), nums.end());
        int i=0;
        int j=nums.size()-1;
        set<int> s;
        while(i<j)
        {
            s.insert(nums[i]+nums[j]);
            i++;
            j--;
        }
        return s.size();
    }
};